doc/api/next_api_changes/behavior/27744-FM.rst

```diff
@@ -0,0 +1,6 @@
+``NavigationToolbar2.save_figure`` now returns filepath of saved figure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``NavigationToolbar2.save_figure`` function return the filename of the saved figure.
+If no figure is save then it returns None.
+For GTK4 backend and Web backend the still returns None.
```

Comment:
```suggestion
``NavigationToolbar2.save_figure`` function may return the filename of the saved figure.  

If a backend implements this
functinoality it should return `None` in the case where no
figure is actually saved (because the user closed the dialog without saving).

If the backend does not or can not implement this functionality (currently the Gtk4 backends and webagg backends
do not) this this method will return `None` as it currently does.
```

Reply:
I tried, `dialog.show()` does not block the function. I think it is because the backend uses a FileChooserNative and not a FileDialog.